Building gtest causes a warning related to CFBundleIdentifier

kkinnunen-apple opened the pull request:
#### 44d1da6908643128a74d810b046e68cc01843fc5
<pre>
Building gtest causes a warning related to CFBundleIdentifier
<a href="https://bugs.webkit.org/show_bug.cgi?id=266404">https://bugs.webkit.org/show_bug.cgi?id=266404</a>
<a href="https://rdar.apple.com/119660379">rdar://119660379</a>

Reviewed by Elliott Williams.

warning: User-supplied CFBundleIdentifier value &apos;com.google.gtest&apos; in
the Info.plist must be the same as the PRODUCT_BUNDLE_IDENTIFIER build
setting value &apos;&apos;.

Add the bundle identifier to the build configuration and use it in
Info.plist.

* Source/ThirdParty/gtest/xcode/Config/General.xcconfig:
* Source/ThirdParty/gtest/xcode/Resources/Info.plist:

Canonical link: <a href="https://commits.webkit.org/272097@main">https://commits.webkit.org/272097@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/9698813388160bfbe50ce1d778da33ef137cfc4a

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/30514 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/9210 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/32177 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/33026 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/27601 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/11492 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/6464 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/27533 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/30822 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/7724 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/27342 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/6616 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/6771 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/27195 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/34367 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/27794 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/27690 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/32942 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/6773 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/4897 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/30765 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/8510 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/7242 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/7505 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/7326 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

emw-apple commented on Source/ThirdParty/gtest/xcode/Config/General.xcconfig:
```diff
@@ -63,5 +63,7 @@ COMBINE_HIDPI_IMAGES = YES;
 // VERSIONING BUILD SETTINGS (used in Info.plist)
 GTEST_VERSIONINFO_ABOUT =  © 2008 Google Inc.
 
+PRODUCT_BUNDLE_IDENTIFIER = com.apple.$(PRODUCT_NAME:rfc1034identifier);
```
FYI, I see that we have both `org.webkit.` and `com.apple.` bundle ID domains in tools code. There doesn't seem to be a rigid pattern, but it might be nice to use `org.webkit.` to mark projects that _aren't_ submitted to the OS. 
```suggestion
PRODUCT_BUNDLE_IDENTIFIER = org.webkit.$(PRODUCT_NAME:rfc1034identifier);
```

emw-apple reviewed (approved):
Makes sense, thank you for fixing!

kkinnunen-apple committed:
Building gtest causes a warning related to CFBundleIdentifier
https://bugs.webkit.org/show_bug.cgi?id=266404
rdar://119660379

Reviewed by Elliott Williams.

warning: User-supplied CFBundleIdentifier value 'com.google.gtest' in
the Info.plist must be the same as the PRODUCT_BUNDLE_IDENTIFIER build
setting value ''.

Add the bundle identifier to the build configuration and use it in
Info.plist.

* Source/ThirdParty/gtest/xcode/Config/General.xcconfig:
* Source/ThirdParty/gtest/xcode/Resources/Info.plist:

Canonical link: https://commits.webkit.org/272097@main
